### DISCUSSION TITLE
11.0.0 Preview 6

### DISCUSSION THREAD

**Discussion by danwalmsley** [AvaloniaUI/Avalonia#10729] [General]
Updated list of breaking changes https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes#xamlnamereferencegenerator-is-part-of-the-main-package-now

<hr /><em>This discussion was created from the release <a href='https://github.com/AvaloniaUI/Avalonia/releases/tag/11.0.0-preview6'>11.0.0 Preview 6</a>.</em>

**Comment by npolyak** [comment]
I imagine in 2050 11.0.0 Preview 246 will be released:) Just kidding. Good job guys, looking forward to RC!!!

**Reply by maxkatz6** [reply]
0.10 also had 6 previews. I wonder if we will stop on it :)

**Comment by kyuranger** [comment]
Can anyone tell us what does Preview 6 changes or adds? Uno Platform team published its 2nd release today and I am looking forward to the Avalonia team publishing an RC version in the future.

**Reply by dotNET-anykey** [reply]
Yeah, would be nice to see changes

**Comment by maxkatz6** [comment]
Updated list of breaking changes from this preview, link in the top message

**Comment by HCareLou** [comment]
I built a program using AvaloniaUI from this version, but the program cannot run. When I try to create a new window, it reports an ambiguity error for the InitializeComponent method.

**Reply by maxkatz6** [reply]
https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes#xamlnamereferencegenerator-is-part-of-the-main-package-now

**Reply by HCareLou** [reply]
Thanks

**Reply by Sigrec** [reply]
> https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes#xamlnamereferencegenerator-is-part-of-the-main-package-now

@maxkatz6 I removed the reference to the package in my .csproj file but it still is saying **error CS0103: The name 'InitializeComponent' does not exist in the current context** for all classes. Is there anything that I am missing here? For some reason when I add the xamlnamegen package back it allows me to build but I get Ambiguity error. I went from preview4 to preview6.

**RESOLVED BY UPDATING VISUAL STUDIO**

**Reply by NitroxydeX** [reply]
Having the same problem right now.

Using Rider the famous error CS0103: The name 'InitializeComponent' does not exist in the current context occures. If this is fixed it can't find any other name in the current context. Is there something else that needs to be done here?

Updated from preview5 to preview6

**Reply by Sigrec** [reply]
It is some weird issue with how project is built it is still referencing your old package ref, idk why updating Visual Studio fixed it though.

**Reply by maxkatz6** [reply]
Or you can update .NET SDK or use nightly

**Reply by KyodaiKen** [reply]
I updated Visual Studio and it still doesn't work for me. I upgrade the nuget packages. Was the biggest mistake ever...

**Reply by timunie** [reply]
Huh? Have you read the upgrade guide carefully?

**Reply by KyodaiKen** [reply]
I wasn't shown any upgrade guide when updating the packages.

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/stay-up-to-date/upgrade-from-0.10

**Reply by KyodaiKen** [reply]
Thanks. I rolled back my project using git. May I ask why there are so many code breaking changes all the time in Avalonia?

**Reply by maxkatz6** [reply]
It’s not true.

For the two years of 0.10 existence we didn’t have any breaking changes, with a goal that library compiled for 0.10.0 would work with application used with 0.10.22 two years later. 

Last summer we have got a new major version - 11.0 - that includes breaking changes. And so far our API was stable and we plan on continuing this trend of stable APIs.

**Comment by chenjing1294** [comment]
where is `IStorageFile.CanOpenWrite` ?

**Reply by maxkatz6** [reply]
Before it always returned true on all platforms. Was removed.
Note, before it didn't check for the permissions, so you still can get an exception, if user denied write access.

**Comment by Sigrec** [comment]
Was LineHeight changed in anyway from 0.10 to 11, When setting a low LineHeight it doesn't apply only large values apply?

**Reply by Gillibald** [reply]
Not intentionally

**Reply by Sigrec** [reply]
I'll look into it more but is there anything related to LineHeight that I should know in terms of how it works with other properties? Also my styles got broken on Button and ComboBox control in other windows as well from my Style xaml class, where I am using a compiled binding and a converter it does not apply on some of the properties mainly when hovering, but it works when using hardcoded enum color (Might be some nesting/tree issue). Everything works fine in my main window though and last thing, SelectedIndex and SelectedItem are not working for me for ComboBox's as well and they are both bound to a string[].

**Reply by Gillibald** [reply]
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Media/TextFormatting/TextLineImpl.cs#L1505
